airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/CdcPostgresSourcePgoutputTest.java

```diff
@@ -4,11 +4,97 @@
 
 package io.airbyte.integrations.source.postgres;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.ImmutableMap;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.commons.util.AutoCloseableIterator;
+import io.airbyte.commons.util.AutoCloseableIterators;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteRecordMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+
 class CdcPostgresSourcePgoutputTest extends CdcPostgresSourceTest {
 
   @Override
   protected String getPluginName() {
     return "pgoutput";
   }
 
+
+  @Override
+  @Test
+  public void testRecordsProducedDuringAndAfterSync() throws Exception {
+
+    final int recordsToCreate = 20;
+    final int[] recordsCreated = {0};
```

Comment:
nitpick: why does this need to be a 1-element array, rather than just `int recordsCreated`?

also, could the `while` loops just be for-loops?